externs/w3c_encoding.js

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2015 The Closure Compiler Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Definitions for W3C's Encoding specification
+ *     https://encoding.spec.whatwg.org
+ * @externs
+ */
+
+/**
+ * @constructor
+ * @param {!string=} encoding
+ * @param {Object=} options
```

Comment:
You probably wouldn't bother doing that, you'd just write `new TextEncoder("UTF8").encode("foo")`. But I guess it's more correct this way so that's fine.

Small nit: Leave off the '!' on string/boolean since it's implied there. Then we should be good to go!